### PR TITLE
[Doc][Dashboard] Add npm ci in docs

### PR DESCRIPTION
**Pull request by marcelocarmona**
## Why are these changes needed?
These changes are added to help users understand how to set up the project environment correctly.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: dashboard/client/README.rst [hunk 1]
```diff
@@ -6,6 +6,12 @@ Available Scripts
 
 In the project directory, you can run:
 
+``npm install``
```
Comment (aslonnie) [inline dashboard/client/README.rst:9]: this should be `npm ci`

File: dashboard/client/README.rst [hunk 2]
```diff
@@ -6,6 +6,12 @@ Available Scripts
 
 In the project directory, you can run:
 
+``npm ci``
+~~~~~~~~~~~~~~~
+
+Once you've run this command, all the dependencies listed in your package.json
+file will be installed.
```
Comment (aslonnie) [inline dashboard/client/README.rst:13]: maybe say installed based on ``package-lock.json``

File: dashboard/client/README.rst [hunk 3]
```diff
@@ -6,6 +6,12 @@ Available Scripts
 
 In the project directory, you can run:
 
+``npm ci``
+~~~~~~~~~~~~~~~
+
+Once you've run this command, all the dependencies listed in your package.json
```
Comment (aslonnie) [inline dashboard/client/README.rst:12]: quote

````
``package.json``
````

?

Comment (marcelocarmona) [reply dashboard/client/README.rst]: I used the same format that we have for

- [``npm start``](https://github.com/ray-project/ray/pull/39123/files#diff-9bc2a03dfd65ef78f0e0d3154f51ee9fed01634f1e90c0f6f14f5a1611d4528cL9)
- [``npm test``](https://github.com/ray-project/ray/pull/39123/files#diff-9bc2a03dfd65ef78f0e0d3154f51ee9fed01634f1e90c0f6f14f5a1611d4528cR24)
- [``npm run build``](https://github.com/ray-project/ray/pull/39123/files#diff-9bc2a03dfd65ef78f0e0d3154f51ee9fed01634f1e90c0f6f14f5a1611d4528cR32)
- [``npm run eject``](https://github.com/ray-project/ray/pull/39123/files#diff-9bc2a03dfd65ef78f0e0d3154f51ee9fed01634f1e90c0f6f14f5a1611d4528cL39)

We are using `inline literal`, if we change that we should update the rest of the commands

Comment (aslonnie) [reply dashboard/client/README.rst]: on https://github.com/pypa/manylinux

it says `manylinux_x_y` requires `pip >= 20.3`, which is `ALT Linux 10+, RHEL 9+, Debian 11+, Fedora 34+, Mageia 8+, Photon OS 3.0 with updates, Ubuntu 21.04+`

File: dashboard/client/README.rst [hunk 4]
```diff
@@ -6,6 +6,11 @@ Available Scripts
 
 In the project directory, you can run:
 
+``npm ci``
+~~~~~~~~~~~~~~~
+
+Installed based on package-lock.json
```
Comment (aslonnie) [inline dashboard/client/README.rst:12]:
```suggestion
Once you've run this command, all the dependencies listed in your package.json are installed based on the versions in package-lock.json
```

File: dashboard/client/README.rst [hunk 5]
```diff
@@ -6,6 +6,11 @@ Available Scripts
 
 In the project directory, you can run:
 
+``npm ci``
+~~~~~~~~~~~~~~~
+
+Once you run this command, all the dependencies listed in your package.json are installed based on the versions in package-lock.json
```
Comment (aslonnie) [inline dashboard/client/README.rst:12]:
```suggestion

``npm ci``
~~~~~~~~~~

Once you run this command, all the dependencies listed in your package.json are installed based on the versions in package-lock.json file.
```

**Review by aslonnie — APPROVED** [review]

**Pull request merged**